contrib/git.lua

```diff
@@ -0,0 +1,245 @@
+  local result
+  local cmd
+  local f
+  local output
+
+--	dt.print_log(path .. " / " .. filename)
+  result = os.execute("git -C " .. path .. " rev-parse --git-dir 2>&1 > /dev/null 2>&1 ")
+  if result == nil then
+    return "No Git repo"
+  end
+
+  cmd = "git -C " .. path .. " status --porcelain --untracked-files=all --ignored " .. filename .. " | cut -b 1-2"
+--	print(" ", cmd)
+  f = io.popen(cmd)
+
+  for line in f:lines() do
+    output = line
+--		print(" ", path.."/"..filename, line)
+  end -- for loop
+  f:close()
+
+  if output == nil then
+    return "Tracked/clean"
+  else
+    if output == " M" then
+      return "Tracked/modified"
+    elseif output == "D " then
+      return "Tracked/del pending"
+    elseif output == "A " then
+      return "Tracked/add pending"
+    elseif output == "??" then
+      return "Not tracked"
+    elseif output == "!!" then
+      return "Ignored"
+    else
+      return "Unknown"
+    end
+  end
+end
+
+-- For a given image, and descriptive git status, set the color
+local function image_set_color_flag(image, gitStatus)
+    if gitStatus == "Tracked/clean" then
+        image.green  = true
+        image.red    = false
+        image.yellow = false
+    elseif gitStatus == "Tracked/add pending" or gitStatus == "Tracked/modified" then
+        image.green  = false
+        image.red    = false
+        image.yellow = true
+    else
+        image.green  = false
+        image.red    = true
+        image.yellow = false
+    end
+end
+
+-- Get the Git status for an image, return descriptive string, set the color
+local function image_update_git_info(image)
+  local imageOutput
+
+  if not image then
+    return "No Image"
+  end
+
+  imageOutput   = git_status_of_file(image.path, image.filename)
+  image_set_color_flag(image, imageOutput)
+
+  return imageOutput
+end
+
+-- Get the Git status for a sidecar, return descriptive string
+local function sidecar_update_git_info(image)
+  local sidecarOutput
+
+  if not image then
+    return "No Image"
+  end
+
+  sidecarOutput = git_status_of_file(image.path, image.sidecar)
+
+  return sidecarOutput
+end
+
+-- register functions to display Git info when mouse-over in lighttable view
+dt.gui.libs.metadata_view.register_info("Git/image", image_update_git_info)
+dt.gui.libs.metadata_view.register_info("Git/sidecar", sidecar_update_git_info)
+
+
+-- Right sidebar GUI to add/remove images from VCS
+local function vcs_action_git(action)
+  dt.print_log("vcs_action_git", action)
+  local sel_images = dt.gui.action_images
+  local sidecarPath
+  local sidecarFilename
+  local result
+
+  for _,image in ipairs(sel_images) do
+    sidecarPath, sidecarFilename = string.match(image.sidecar, "(.-)([^/]-([^%.]+))$")
+
+    dt.print_log(" ", image.path .. " / " .. image.filename)
+    dt.print_log(" ", sidecarPath .. " / " .. sidecarFilename)
+
+    result = os.execute("git -C " .. image.path .. " " .. action .. " " .. image.filename)
```

Comment:
Have you tested this with filenames with spaces? we have a the utility function df.sanitize_filename() in "lib/dtutils.file". Perhaps you can use it here.